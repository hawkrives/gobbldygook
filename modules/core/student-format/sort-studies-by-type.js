import sortBy from 'lodash/sortBy'

const types = [ 'degree', 'major', 'concentration', 'emphasis' ]
export function sortStudiesByType(studies) {
	return sortBy(studies, s => types.indexOf(s.type))
}
