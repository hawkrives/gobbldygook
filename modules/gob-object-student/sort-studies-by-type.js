// @flow
import sortBy from 'lodash/sortBy'

import {type AreaQuery} from './types'

const types = ['degree', 'major', 'concentration', 'emphasis']
export function sortStudiesByType(studies: Array<AreaQuery>) {
	return sortBy(studies, s => types.indexOf(s.type))
}
