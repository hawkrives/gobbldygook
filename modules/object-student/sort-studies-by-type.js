// @flow
'use strict'
const sortBy = require('lodash/sortBy')

const types = ['degree', 'major', 'concentration', 'emphasis']
function sortStudiesByType(studies: string[]) {
    return sortBy(studies, s => types.indexOf(s.type))
}

module.exports.sortStudiesByType = sortStudiesByType
