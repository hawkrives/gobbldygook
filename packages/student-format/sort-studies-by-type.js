// @flow

import sortBy from 'lodash/sortBy'
import type {Study, StudyTypes} from '../models/types'

const types: StudyTypes[] = ['degree', 'major', 'concentration', 'emphasis']
export default function sortStudiesByType(studies: Study[]): Study[] {
	return sortBy(studies, (s: Study) => types.indexOf(s.type))
}
