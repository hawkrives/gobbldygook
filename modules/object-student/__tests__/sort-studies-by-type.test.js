import { sortStudiesByType } from '../sort-studies-by-type'

describe('sortStudiesByType', () => {
    it('sorts a list of areas of study by type', () => {
        const input = [
            { type: 'degree' },
            { type: 'concentration' },
            { type: 'emphasis' },
            { type: 'major' },
        ]

        expect(sortStudiesByType(input)).toMatchSnapshot()
    })
})
