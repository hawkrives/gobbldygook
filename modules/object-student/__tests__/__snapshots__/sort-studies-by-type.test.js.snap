// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sortStudiesByType sorts a list of areas of study by type 1`] = `
Array [
  Object {
    "type": "degree",
  },
  Object {
    "type": "major",
  },
  Object {
    "type": "concentration",
  },
  Object {
    "type": "emphasis",
  },
]
`;
